webrender/src/frame.rs

```diff
@@ -320,6 +320,10 @@ impl Frame {
                                                   background_color,
                                                   self.frame_builder_config);
 
+        if let Some(ref mut old_builder) = self.frame_builder {
```

Comment:
Another approach that could be cleaner is to pass `self.frame_builder` as a parameter into `FrameBuilder::new`, which would just take & clear the vectors it needs from the old one. That would make sure no items are left behind (without `recycle_vec`, which feels a bit hacky to me).